test/distributed/_sharded_tensor/test_megatron_prototype.py

```diff
@@ -0,0 +1,119 @@
+# Owner(s): ["oncall: distributed"]
+
+import sys
+
+import torch
+import torch.distributed as dist
+from torch.distributed._sharded_tensor import (
+    aggregate_partial_tensor_list,
+    shard_parameter,
+)
+from torch.testing._internal.common_distributed import (
+    requires_nccl,
+    skip_if_lt_x_gpu,
+)
+from torch.testing._internal.common_utils import (
+    TEST_WITH_DEV_DBG_ASAN,
+    run_tests,
+)
+from torch.testing._internal.distributed._sharded_tensor import (
+    TEST_GPU_NUM,
+    ShardedTensorTestBase,
+    with_comms,
+)
+from torch.testing._internal.distributed._sharded_tensor._test_ops_common import (
+    generate_chunk_sharding_specs_for_test,
+    generate_local_weight_sharding_params_for_test,
+)
+
+if TEST_WITH_DEV_DBG_ASAN:
+    print(
+        "Skip dev-asan as torch + multiprocessing spawn have known issues",
+        file=sys.stderr,
+    )
+    sys.exit(0)
+
+
+class TestShardedTensorMegatronLinear(ShardedTensorTestBase):
+    def _run_megatron_linear(self, spec, input_size, linear_size):
+        # Use same seed.
+        torch.manual_seed(0)
+        linear_size_first = linear_size[0]
+        local_linear_first_layer = torch.nn.Linear(*linear_size_first).cuda(self.rank)
+        sharded_linear_first_layer = torch.nn.Linear(*linear_size_first)
+
+        torch.manual_seed(1024)
+        linear_size_second = linear_size[1]
+        local_linear_second_layer = torch.nn.Linear(*linear_size_second).cuda(self.rank)
+        sharded_linear_second_layer = torch.nn.Linear(*linear_size_second)
+
+        # Copy the weights and bias from local linear
+        sharded_linear_first_layer.weight = torch.nn.Parameter(local_linear_first_layer.weight.detach().clone())
```

Comment:
the param copy between sharded_model and a local model seems pretty popular in different tests, maybe we can make it a test util?